### ISSUE TITLE
Delete Data objects

### ISSUE THREAD

**Issue by noheton** [noheton/shepard#319]
***In GitLab by @wiglef on Nov 20, 2024, 16:41 GMT+1:***

# Feature Request

## :goal_net: Goal

As a frontend user I want to be able to delete data objects in the frontend so that I can get rid of unused elements.

## :sports_medal: Acceptance Criteria

1. Delete functionality is hidden in a context menu. Delete functionality runs the "soft delete" functionality that the backend currently implements
2. There is a confirmation dialogue 

## :star: Visual Design

https://www.figma.com/design/ivBPq71AuUl2ohFnrzHIKF/Visual-Design?node-id=974-15406&m=dev

## :bomb: Out of Scope

- Option to delete with all of its children is a desired functionality that can be added later
- in case of deletion of a parent, it would be more intuitive for the user if the children then inherit their grandparent instead of having no parent at all, not strictly necessairy for MVP

## :thinking_face: Risks & Open Questions

**Assignees:** xit-paul-reinhardt

*Migrated from GitLab: https://gitlab.com/dlr-shepard/shepard/-/work_items/319*

**Comment by noheton** [comment]
***In GitLab by @wiglef on Nov 20, 2024, 16:41 GMT+1:***

added to epic &8

**Comment by noheton** [comment]
***In GitLab by @xit-luk-wer on Jan 15, 2025, 15:50 GMT+1:***

mentioned in commit 5806db7cb1c6c6dd37362068ee86c55db4f632ce

**Comment by noheton** [comment]
***In GitLab by @xit-luk-wer on Jan 15, 2025, 16:06 GMT+1:***

mentioned in commit 8b95d23f7fa2ce98ffb4be323b91edffdfa23aa0

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jan 16, 2025, 13:17 GMT+1:***

mentioned in commit 7f189637a7778b8e27a06fc6017d0840b40b5696

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jan 16, 2025, 13:26 GMT+1:***

mentioned in commit cccb1dd242c99ca1d1189a20dc34f793a78ae15c

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jan 16, 2025, 13:31 GMT+1:***

added #404 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jan 16, 2025, 13:43 GMT+1:***

mentioned in commit f1a06e7e6a19d42a36ed21906efb711c6d408435

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jan 17, 2025, 08:29 GMT+1:***

marked this issue as blocked by #339

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jan 17, 2025, 08:29 GMT+1:***

The only open issue is #319. For this I am waiting for the sidebar refactoring in #339

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jan 20, 2025, 17:36 GMT+1:***

We fixed it together with #339

**Comment by noheton** [comment]
***In GitLab by @wiglef on Jan 21, 2025, 09:18 GMT+1:***

removed the relation with #339

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jan 21, 2025, 10:54 GMT+1:***

Testing:

* When deleting a group item with it's children not loaded, the children are not added to top level
* When deleting a group item and the children have been added to top level, deleting on of the previous children does not update the treeview properly
* The order of the updated top level treeview items does not match the order when refreshing the page

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jan 21, 2025, 13:15 GMT+1:***

mentioned in commit c2b0223c8930226b5d1f66471f7f2a2e09b672ca

**Comment by noheton** [comment]
***In GitLab by @wiglef on Jan 23, 2025, 10:00 GMT+1:***

We forgot to specify that the delete button should only show up for users that actually have the permission to do so, but we should implement it here as we want to have that for edit as well. I'll add an AC

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jan 23, 2025, 10:49 GMT+1:***

We will implement this with the edit data object & collections story anyway. I would prefer if you add the AC there so that this story can be closed.
